src/security-council-mgmt/SecurityCouncilManager.sol

```diff
@@ -0,0 +1,292 @@
+    /// @param _newMember member to add
+    /// @param _cohort cohort to add member to
+    function addMemberToCohort(address _newMember, Cohort _cohort)
+        external
+        onlyRole(MEMBER_ADDER_ROLE)
+    {
+        address[] storage cohort = _cohort == Cohort.MARCH ? marchCohort : septemberCohort;
+        require(cohort.length < 6, "SecurityCouncilManager: cohort is full");
+        require(
+            !SecurityCouncilMgmtUtils.isInArray(_newMember, marchCohort),
+            "SecurityCouncilManager: member already in march cohort"
+        );
+        require(
+            !SecurityCouncilMgmtUtils.isInArray(_newMember, septemberCohort),
+            "SecurityCouncilManager: member already in septemberCohort cohort"
+        );
+
+        cohort.push(_newMember);
+
+        address[] memory membersToAdd;
+        membersToAdd[0] = (_newMember);
+
+        address[] memory membersToRemove;
+        _dispatchUpdateMembers(membersToAdd, membersToRemove);
+        emit MemberAdded(_newMember, _cohort);
+    }
+
+    /// @notice callable only by SC Removal Governor.
+    /// Don't need to specify cohort since duplicate members aren't allowed (so always unambiguous)
+    /// @param _member member to remove
+    function removeMember(address _member) external onlyRole(MEMBER_REMOVER_ROLE) returns (bool) {
+        if (_removeMemberFromCohort(_member, marchCohort)) {
+            emit MemberRemoved(_member, Cohort.MARCH);
+            return true;
+        }
+        if (_removeMemberFromCohort(_member, septemberCohort)) {
+            emit MemberRemoved(_member, Cohort.SEPTEMBER);
+            return true;
+        }
+
+        revert("SecurityCouncilManager: member not found");
+    }
+
+    /// @notice Removes member in this contract's state and triggers dispatch
+    /// @param _member member to remove
+    /// @param _cohort cohort to remove member from
+    function _removeMemberFromCohort(address _member, address[] storage _cohort)
+        internal
+        returns (bool)
+    {
+        for (uint256 i = 0; i < _cohort.length; i++) {
+            if (_member == _cohort[i]) {
+                delete _cohort[i];
+                address[] memory membersToAdd;
+                address[] memory membersToRemove;
+                membersToRemove[0] = _member;
+                _dispatchUpdateMembers(membersToAdd, membersToRemove);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /// @notice Security council member can rotate out their address for a new one.
+    /// Rotation must be initiated by the security council, and member rotating out must give explicit
+    /// consent via signature
+    /// @param _currentAddress Address to rotate out
+    /// @param _newAddress Address to rotate in
+    /// @param _signature Signature from _currentAddress
+    function rotateMember(address _currentAddress, address _newAddress, bytes memory _signature)
```

Comment:
re the later, we want to make sure this method doesn't grant the SC powers it wouldn't otherwise have; in the case of a member losing their keys, replacement should happen via the "remove member" path.

re the former... good point lol. this was discussed but i can't quite remember the rationale now, will raise it with the SC

Reply:
update: not using msg.sender b/c we want to include cases where the address's owner lost access to their keys